packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -383,9 +398,13 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         bytes memory _data
     )
         public
-        payable
         metered(_gasLimit)
     {
+        // Lock token in this contract
+        if (_value > 0) {
+            IERC20(tonAddress).safeTransferFrom(msg.sender, address(this), _value);
+        }
+
```

Comment:
are you locking ton to optimismPortal?? If so, I have a question for you.
I know that L1CrossDomainMessenger calls optimismPortal. so this msg.sender is L1CrossDomainMessenger. 
is right?

and L1StandardBridge also has logic to send ton to L1bridge. This means that L1CrossDomainMessenger has never received a ton.

but above logic is L1CrossDomainMessenger send ton to OptimismPortal

I don't understand, I'm not very experienced with smart contracts so maybe I'm just not understanding the code.

please help

Reply:
Thank you!
L1CrossDomainMessenger  also receives TON when the function sendTONMessage is called. 
Anyone can call depositTransaction, in case users deposit TON by using L1StandardBridge, the flow is below:
 L1StandardBridge -> L1CrossDomainMessenger -> OptimismPortal.

Reply:
> Thank you! L1CrossDomainMessenger also receives TON when the function sendTONMessage is called. Anyone can call depositTransaction, in the case msg.sender that you mentioned, it can by any address (not only L1CrossDomainMessenger). When users deposit TON by using L1StandardBridge, the flow is below: L1StandardBridge -> L1CrossDomainMessenger -> OptimismPortal.